### PR TITLE
basic SHOCK functionality for mock_kbase image

### PR DESCRIPTION
**Pull request by mlhenderson**
This initial version contains a running MongoDB and SHOCK instance.  Only the SHOCK instance has a port exposed currently.

supervisord is used to fire up mongod and shock-server when the container starts up.

A .travis.yml file is included, which tests the docker build, and posts a random 1GB file to SHOCK.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10 && \
+    echo 'deb http://downloads-distro.mongodb.org/repo/ubuntu-upstart dist 10gen' | sudo tee /etc/apt/sources.list.d/mongodb.list && \
+    apt-get update && \
+    apt-get install -y \
+        libsasl2-dev libsasl2-modules-gssapi-mit \
+        mongodb-10gen=2.4.14 \
+        supervisor \
+        expect
+
+# update golang to 1.3.x
```
Comment (sychan) [inline Dockerfile:20]: Really? Shock is only at Go 1.3? Current release is 1.7! Or is KBase just running on a version of Shock that only runs on 1.3?

Comment (mlhenderson) [reply Dockerfile:20]: KBase is running on an older version of Shock.  The develop branch of Shock requires 1.6, but it is unclear what that will break.  Keith and I discussed this briefly, and I was going to test with golang 1.6 to see if anything broke.  However, I won't be sure until the other components are integrated, so I was going to wait until then and try a newer version of golang.  Newer versions of Shock may be required for newer golang versions, and then that may bring API changes, etc.

Anyway, this matches current deployments of KBase.  Happy to test out newer versions of things as well.

Comment (mlhenderson) [reply Dockerfile:20]: Correction, latest Shock is at 1.7.1.  KBase is using a commit that is nearest to the 0.9.6 tagged release of Shock, with the current tag at 0.9.18.

https://github.com/MG-RAST/Shock/blob/master/RELEASE_NOTES.txt

Probably worth testing more recent releases of Shock once this container has the other KBase pieces in place.

**Comment by mlhenderson** [comment]
@scanon correctly pointed out that we will need to deal with the admin user for Shock.  I'll roll that into a future PR.

**Pull request merged**